File: cpp/prob006.cpp
#include <iostream>
#include <chrono>

int compute()
{
    int sum = 0, sq_sum = 0;
    for(int i = 1; i <= 100; i++){
        sum += i;
        sq_sum += i*i;
    }
    return sum*sum - sq_sum;
}

int main(int argc, char const *argv[])
{
    auto t1 = std::chrono::high_resolution_clock::now();
    int64_t result = compute();
    auto t2 = std::chrono::high_resolution_clock::now();

    auto duration = std::chrono::duration_cast<std::chrono::microseconds>(t2 - t1).count();
    std::cout << result << "\tin " << duration << " us\n";
    return 0;
}

File: rust/src/bin/prob006.rs
extern crate euler;

fn compute() -> i32{
    let num: i32 = 100;
    ((1..=num).sum::<i32>() * (1..=num).sum::<i32>()) - (1..=num).fold(0, |acc, x| acc + x*x)
}

fn main() {
    let now = std::time::Instant::now();
    let result = compute();
    println!("{:?}\tin {}us", result, now.elapsed().as_micros());
}


#[cfg(test)]
mod tests {
    use super::*;
    const PROB_NUM: usize = 6;
    
    #[test]
    fn test_prob006() {
        let result = compute().to_string();
        euler::test_answer(PROB_NUM, result);
    }
}